### DISCUSSION TITLE
`AttributeError` thrown when `validate_assignment=True`

### DISCUSSION THREAD

**Discussion by Afstkla** [fastapi/sqlmodel#763] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pydantic import ConfigDict
from sqlmodel import Field, SQLModel


class Hero(SQLModel, table=True):
    model_config = ConfigDict(validate_assignment=True)

    id: int = Field(primary_key=True)


h1 = Hero(id=1)  # error happens here
```

### Description

I've just upgraded my code to SQLModel 0.0.14 (and Pydantic 2.5, SQLAlchemy 2.0.23). However, when I run the minimal example above, I immediately get the following error:

```
'Hero' object has no attribute '__pydantic_extra__'
  File "scratch.py", line 11, in <module>
    h1 = Hero(id=1)
         ^^^^^^^^^^
AttributeError: 'Hero' object has no attribute '__pydantic_extra__'
```

When I apply the patch [here](https://github.com/NeopleLabs/sqlmodel/commit/8f504860e3bb816ad60a6a6b00bfa59135e2ee3e), it solves the issue in the example. However, in a longer example like below, specifically when getting models from the database, I still get the same `__pydantic_extra__` error:

```python
from pydantic import ConfigDict
from sqlmodel import Field, Session, SQLModel, create_engine, select


class Hero(SQLModel, table=True):
    model_config = ConfigDict(validate_assignment=True)
    id: int = Field(primary_key=True)
    name: str


engine = create_engine("sqlite+pysqlite:///:memory:")

SQLModel.metadata.create_all(engine)

h1 = Hero(name="a")
with Session(engine) as s:
    s.add(h1)
    s.commit()

with Session(engine) as s:
    hero = s.exec(select(Hero).where(Hero.name == "a")).first()

hero.name = "b" # error happens here
```

Does anyone have any ideas of what I could try, what I'm doing wrong, or where I should look to find the source and submit a PR?

### Operating System

macOS

### SQLModel Version

0.0.14

### Python Version

3.11.6

### Additional Context

Pydantic 2.5.3
SQLAlchemy 2.0.23

**Comment by YuriiMotov** [comment]
Works well in current version (0.0.24)
